### PR TITLE
Improve cmake install path for extraction tools

### PR DESCRIPTION
**Pull request by schell244**
## 🍰 Pullrequest
Copy map extraction tools into `${BIN_DIR}/Extractors` folder, also copy `offmesh.txt` and `config.json` to the same location.
Added a more consistent naming scheme for the tools:

| Old Name         | New Name           |
|------------------|--------------------|
| `mapextractor`   | `MapExtractor`     |
| `MoveMapGen`     | `MoveMapGenerator` |
| `vmap_assembler` | `VMapAssembler`    |
| `vmapextractor`  | `VMapExtractor`    |

### How to test
- Compile with `USE_EXTRACTORS = 1`
- CMakePredefinedTargets -> INSTALL
- map extraction tools will be copied into Extractors folder

### CODE REVIEW

**Comment by mserajnik** [comment]
The executable names (`MoveMapGen` and `MoveMapGen.exe`) will have do be adjusted in this script too then: https://github.com/vmangos/core/blob/development/contrib/mmap/mmap_extract.py

File: contrib/extractor/CMakeLists.txt [hunk 1]
```diff
@@ -23,24 +23,27 @@ include_directories (
...
   endif(ZLIB_FOUND)
 else()
-  target_link_libraries (mapextractor zlib libmpq bz2)
+  target_link_libraries (MapExtractor zlib libmpq bz2)
 endif(UNIX)
 
-install(TARGETS mapextractor DESTINATION ${BIN_DIR})
+install(TARGETS MapExtractor DESTINATION ${BIN_DIR}/Extractors)
+if(WIN32 AND MSVC)
+  install(FILES ${CMAKE_SOURCE_DIR}/bin/Debug/MapExtractor.pdb DESTINATION ${BIN_DIR}/Extractors CONFIGURATIONS Debug)
```
Comment (0blu) [inline contrib/extractor/CMakeLists.txt]: `BIN_DIR` is correct (which is the install path) but you have to adjust the source, use `${CMAKE_BINARY_DIR}` for that.

File: contrib/mmap/CMakeLists.txt [hunk 2]
```diff
@@ -59,4 +59,10 @@ target_link_libraries(MoveMapGen
                         framework
                         )
 
-install(TARGETS MoveMapGen DESTINATION ${BIN_DIR})
+install(TARGETS MoveMapGenerator DESTINATION ${BIN_DIR}/Extractors)
+install(FILES ${CMAKE_SOURCE_DIR}/contrib/mmap/offmesh.txt DESTINATION ${BIN_DIR}/Extractors)
+install(FILES ${CMAKE_SOURCE_DIR}/contrib/mmap/config.json DESTINATION ${BIN_DIR}/Extractors)
+
+if(WIN32 AND MSVC)
+  install(FILES ${CMAKE_SOURCE_DIR}/bin/Debug/MoveMapGenerator.pdb DESTINATION ${BIN_DIR}/Extractors CONFIGURATIONS Debug)
```
Comment (0blu) [inline contrib/mmap/CMakeLists.txt]: Same as above

**Review by 0blu — CHANGES_REQUESTED** [review]

**Review by 0blu — APPROVED** [review]
lgtm! I would squash merge it.

**Review by mserajnik — APPROVED** [review]
Since the script needs to be placed in the same directory as the `MoveMapGenerator` executable to work anyway, this seems like a good addition to me so the users don't have to do that manually.

__Edit:__ Whoops, this was meant to get added as a comment to [this commit](https://github.com/vmangos/core/pull/2803/commits/06fbf182f3e32eb259718e4b71c2f707fa890cab), not sure why it shows as me approving _all_ the changes in the PR instead.

**Comment by mserajnik** [comment]
Hm, how about also renaming `mmap_extract.py` to something more consistent? Maybe `GenerateMoveMaps.py`; or maybe better `generate_move_maps.py` if we want a different style to visually distinguish the script more from the compiled executables. Not sure what would be preferable here.

**Pull request merged**